### ISSUE TITLE
Discussion: UX week thoughts

### ISSUE THREAD

**Issue by ZooeyMiller** [foundersandcoders/master-reference#269]
## UX week discussion points

I have a few thoughts from UX week, I think overall it is a valuable part of the curriculum but feel there are aspects that could be changed.

The main crux of this is I feel dedicating the entire week 10-6pm every day to the design sprint is not a completely efficient use of time. By the end of the days (the last 2 hours or so) a lot of the people were flagging and not really doing that much. 

I feel perhaps dedicating 2 - 3 days to the design sprint and the other days to other bits related to UX would be really beneficial. Or perhaps having the morning each day (either the whole morning or 2 or so hours) dedicated so something else UX-y and not all the design sprint would be beneficial.

The things I would feel would be beneficial:

### Basic UX concepts
I do not feel any more well equipped in terms of what basic UX concepts are or how to implement them. Though UX is obviously a field full of different opinions and does not always have 100% 'right' and 'wrong' ways to do things, there are some relatively well accepted UX concepts and feel teaching us some of these would be beneficial.

### Accessibility
We had a bit of stuff in week one re: accessibility and in the first few projects we had some issues raised about a11y in our projects, but with the emphasis coming away from front end and moving towards back end and styling and UX being something to only be done if there is the time, a lot of things relating to a11y have fallen out of the front of my mind. 
I feel a refresher in UX week would be really really good, and cements a11y as a feature of UX, from the start and not an afterthought.

### Performance

At no point during the course (as far as I am aware) have we had any information on improving performance of web apps, when this is really important. It's a factor of UX, it's a factor of a11y and it's important that as developers we keep performance in mind.
First meaningful paint is a really important aspects of UX (imo), if the user leaves the page before it loads because it's taking too long, that is a UX issue.
It's important to a11y because the majority of people in the world are using mobile devices to use the web and are on 2g connections, and are on less performant devices than the majority of us, in London, and in the tech community in particular. 
It's important to us as developers as it should be part of our thought process whenever we are writing code, if we go our whole path as developers not considering performance, or even __how__ to optimise performance then we will be weaker and the products we create will be worse.

**Comment by jsms90** [comment]
@sofiapoh sounds like it would be good to reintroduce your talk on performance from fac9's ux weeks

**Comment by harrygfox** [comment]
Hi @ZooeyMiller!
Thanks very much for opening this issue - I love that you're passionate about it!
@nogainbar and I have spent time together today restructuring UX week to (we hope) directly address some of the concerns you have raised above [notably around basic UX/UI wins and accessibility & usability].

**Comment by ZooeyMiller** [comment]
Awesome! Thanks @harrygfox 
Really looking forward to FAC11 go through the process.

**Comment by sofiapoh** [comment]
@jsms90 @ZooeyMiller Would be happy to do this, would this be for FAC10 or 11?

**Comment by ZooeyMiller** [comment]
@sofiapoh We'd love to have the talk for FAC10 if you're available! Could you do Tues 30th May or Weds 31st May? A 17:00-18:00 slot? 
I know it's super short notice so no worries if not.

**Comment by sofiapoh** [comment]
@ZooeyMiller let me check my schedule, I could potentially do next week. I'll confirm tomorrow :)

**Comment by jsms90** [comment]
@sofiapoh Get someone to record you! I feel like there was gold in there, and I was really flagging at that point in FAC9 :sweat_smile:

**Comment by ZooeyMiller** [comment]
@sofiapoh awesome! thank you.

**Comment by sofiapoh** [comment]
@ZooeyMiller sorry forgot to confirm but weds works! Is around 5-5:30 okay?

**Comment by ZooeyMiller** [comment]
@sofiapoh Perfect! See you then :)

**Comment by jsms90** [comment]
Just to add some prior context...

the [FAC10 week 10 SGC](https://github.com/foundersandcoders/london-curriculum/blob/master/stop-go-continue/fac-10/cohort-week-10.md):
> #### Go
> + More general info on UX principles and theory
> + Last time there was a talk on ‘Don’t Make Me Think’ and UX/UI – maybe some of that could be brought back but lots of it is very partial.

the [FAC9 weeks 9-10 SGC](https://github.com/foundersandcoders/master-reference/issues/36#issuecomment-268903246) ("UX week"):
> #### Continue
> + Sharing Inspiring sites

(this referred to sites like https://muz.li/ & http://www.material-ui.com/)

and positive feedback from FAC9 & FAC10 about @sofiapoh's talk on performance

But also form the [FAC9 weeks 9-10 SGC](https://github.com/foundersandcoders/master-reference/issues/36#issuecomment-268903246) ("UX week"):
> #### Stop
> + Combination of Project Management AND UX was too confusing
>    + What was the purpose of this week? Was this about UX design or about managing clients and projects? Repackage the week to make it less confusing

**Comment by jsms90** [comment]
@ZooeyMiller hope you don't mind, but I've updated the title.

Please change it if you feel that it doesn't adequately describe what you were trying to convey?

**Comment by ZooeyMiller** [comment]
All good with me! I would say that if it's going to remain in a similar state to how it was for FAC10 then renaming it to something related to like "Design sprint week" or "Intro to agile" week would be more apt, as it stood I didn't feel like it spoke to enough aspects of UX to genuinely be considered a UX week.

**Comment by jsms90** [comment]
@ZooeyMiller Yeah "design thinking" was exactly what @nogainbar & @bradreeder suggested :+1: 

In this repo, it's now sitting inside [the folder for the 3-week project](https://github.com/foundersandcoders/master-reference/tree/master/coursebook/weeks-10-12) under "design sprint" & I'm speaking to FAC11 today to give an understanding of what to expect from the second half. Should be much clearer this time.

I still think it's worth keeping this issue open though, since we don't cover UX much in the course & you've got some good suggestions in the top comment.

**Comment by ibraghada** [comment]
Hey @ZooeyMiller and all here!

Couldn't help but jump in here as I saw the letters UX :D 
Being a FACG1 graduate, I do agree that the curriculum didn't cover enough about that! and something I notice much is that most of the graduates, if not all of them actually neglect that aspect of the things they build. To many, It's all about writing the code and make it work while the whole user experience is crucial to define how good/successful/usable your product is, even more how amazing is the experience that you give to your users!! 

I believe this quote applies here,
>“[User experience] is used by people to say ‘I’m a user experience designer, I design websites’, or ‘I design apps.’ And they have no clue as to what they’re doing, and they think the experience is that simple device, the website, or the app, or who knows what. No! It’s everything—it’s the way you experience the world, it’s the way you experience your life, it’s the way you experience the service. Or, yeah, an app or a computer system. But it’s a system that’s everything.” 
— Don Norman, pioneer and inventor of the term “user experience”.

I've done some design thinking co-teaching for 2 weeks last month with 2 groups of 14-16ish school students learning how to code. And I have introduced (**Interaction Design**) which falls into the spectrum of UX process elements/disciplines, other disciplines include [Information Architecture, Visual Design, Functionality, Usability, User Interface, Typography, Content Strategy] 
UX is not just about satisfying the user's stories/needs, it goes beyond that features checklist! 

You can Check out the presentation here if you like to https://docs.google.com/presentation/d/1SWsoGq5D0eC80eQV-FEuRwscWJI9CIGA3bOHOk8pyJw/edit?usp=sharing

There's also a website called Night Walk that might give you a glimpse of what we mean with "experience". Check it out https://nightwalk.withgoogle.com/en/panorama
